src/mine-sweeper.tsx

```diff
@@ -176,16 +180,40 @@ export default class MineSweeper extends React.Component<IMineSweeperProps, IMin
     }
   }
 
-  private revealCell(x: number, y: number) {
-    const mineField = this.state.mineField;
-    mineField[x][y].isRevealed = true;
-    mineField[x][y].isFlagged = false;
-    this.setState({
-      mineField
-    });
-    this.checkSurroundingCells(x, y);
+  private revealCell = (x: number, y: number) => {
+    const field = this.state.mineField[x][y]
+    this.revealCellByMineField(field)
+    if (field.mineCounter === 0) {
+      this.revealSurroundingCells(x, y)
+    }
   }
 
+  private revealCellByMineField = (cell: IMineField) =>
+  {
+    if (cell.isMine) { return }
+    else {
+      cell.isRevealed   = true
+      cell.isFlagged    = false
+      this.updateState()
+    }
+  }
+
+  private updateState = () => this.setState({})
```

Comment:
you are not passing anything to setState though?

Reply:
yeah it just triggers the update to the UI, it works as an update